server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -444,6 +450,20 @@ public long sizeOfGensAboveSeqNoInBytes(long minSeqNo) {
         }
     }
 
+    private long earliestGenerationLastModifiedAge(long minGeneration) {
```

Comment:
I think we can reduce complexity here as we don't need min generation control. Can we remove the parameter please?

Reply:
maybe make this method static, give it a list/stream of readers and the current time as parameters? this would allow for proper testing with mocks and fixed time.

Reply:
A static method for testing is a good idea. If we made it static we would still need to wrap it in a non static method to include:

```java
try (ReleasableLock ignored = readLock.acquire()) {
    ensureOpen();
    ...
}```

Reply:
sure.